### PR TITLE
ESLint rules configuration updates

### PR DESCRIPTION
**Pull request by edalex-ian**
##### Checklist

- [x] the [contributor license agreement][] is signed
- [x] commit message follows [commit guidelines][]
- [x] tests are included
- [ ] screenshots are included showing significant UI changes
- [ ] documentation is changed or added

##### Description of change

This makes some minor settings updates for ESLint, but most noticeably adds the rules from `eslint:recommended`. This was mostly clear, other than lots of [no-case-declarations](https://eslint.org/docs/rules/no-case-declarations) and [no-prototype-builtins](https://eslint.org/docs/rules/no-prototype-builtins) violations.

Also highlighted a couple of redundant (`encodequery`) or unused (`dictionary.ts`) files. So they were removed too. In one case that meant bringing in [query-string](https://www.npmjs.com/package/query-string). I did consider using the more popular [qs](https://www.npmjs.com/package/qs) however the defaults in query-string were better and we already use query-string in the REST module.

[contributor license agreement]: https://www.apereo.org/node/676
[commit guidelines]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: react-front-end/__tests__/tsrc/search/AdvancedSearchTestHelper.ts [hunk 1]
```diff
@@ -500,7 +500,7 @@ export const updateControlValue = (
         userEvent.type(editBox, value);
       });
       break;
-    case "checkboxgroup":
+    case "checkboxgroup": {
```
Comment (edalex-ian) [inline react-front-end/__tests__/tsrc/search/AdvancedSearchTestHelper.ts:503]: With some of the case statements you'll see I went with simply adding a block. As the refactor was too significant. In some ways though,t his potentially highlights there is too much code in the case statement and should be extracted into one or more functions.

File: react-front-end/tsrc/cloudcontrol/index.ts [hunk 2]
```diff
@@ -271,12 +271,13 @@ export const CloudControl: CloudControlRegisterImpl = {
           case "deleted":
             att = att.filter((at) => at.uuid !== change.uuid);
             break;
-          case "edited":
+          case "edited": {
```
Comment (edalex-ian) [inline react-front-end/tsrc/cloudcontrol/index.ts:274]: This was a 'safety first' approach - being that it's the cloudcontrol code with no test coverage.

File: react-front-end/tsrc/modules/SearchFilterSettingsModule.ts [hunk 3]
```diff
@@ -64,7 +64,7 @@ export const batchUpdateOrAdd = (filters: MimeTypeFilter[]) =>
 
 export const batchDelete = (ids: string[]) =>
   Axios.delete<BatchOperationResponse[]>(
-    `${MIME_TYPE_FILTERS_URL}/${encodeQuery({ ids: ids })}`
+    `${MIME_TYPE_FILTERS_URL}/?${encodeQuery({ ids: ids })}`
```
Comment (edalex-ian) [inline react-front-end/tsrc/modules/SearchFilterSettingsModule.ts:67]: These two blocks of code should actually be over in the REST module. When that is done, perhaps query-string may no longer be needed in the react-front-end. :thinking:

Comment (PenghaiZhang) [reply react-front-end/tsrc/modules/SearchFilterSettingsModule.ts:67]: I remember we still have some tickets about moving this kind of code to REST module in backlog.

**Review by PenghaiZhang — APPROVED** [review]
Thanks Ian! Lots of efforts !

File: react-front-end/__tests__/tsrc/search/AdvancedSearchTestHelper.ts (hunk 1 above)
Comment (edalex-ian) [reply react-front-end/__tests__/tsrc/search/AdvancedSearchTestHelper.ts:503]: Yup, and that makes sense. It's just more so in this situation with a big switch statement with many cases we probably need to focus on the bigger picture and pull them out. But for now, no need to change it. Something for the future/next time.

File: react-front-end/tsrc/modules/SearchFilterSettingsModule.ts (hunk 3 above)
Comment (edalex-ian) [reply react-front-end/tsrc/modules/SearchFilterSettingsModule.ts:67]: Hmmm, good point. I actually raised two new ones, but maybe they're duplicates... :thinking: 

Anyway, we'll tackle in this release.

File: react-front-end/build-tools/BuildLanguageBundle.ts [hunk 4]
```diff
@@ -43,7 +43,7 @@ const generateLanguageBundle = (
     output.push({ [key]: value });
   } else if (typeof value === "object") {
     for (const subKey in value) {
-      if (value.hasOwnProperty(subKey)) {
+      if (Object.prototype.hasOwnProperty.call(value, subKey)) {
```
[the remark this answers is not in the document]
Comment (edalex-ian) [reply react-front-end/build-tools/BuildLanguageBundle.ts:46]: That's this one: [no-prototype-builtins](https://eslint.org/docs/rules/no-prototype-builtins)

**Comment by edalex-ian** [comment]
Cool, just a flake on ScalaCheck on Codebuild.

**Pull request merged**